ledger-core/virtual/integration/method_test.go

```diff
@@ -2097,3 +2098,121 @@ func TestVirtual_Method_CheckValidatedState(t *testing.T) {
+		secondOutgoing       = server.BuildRandomOutgoingWithPulse()
+		synchronizeExecution = synchronization.NewPoint(1)
+	)
+
+	// add ExecutionMock to runnerMock
+	{
+		runnerMock.AddExecutionMock(firstOutgoing.String()).AddStart(
+			func(ctx execution.Context) {
+				assert.Equal(t, objectGlobal, ctx.Object)
+				assert.Equal(t, contract.CallIntolerable, ctx.Isolation.Interference)
+				synchronizeExecution.Synchronize()
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("first result"), objectGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionMock(secondOutgoing.String()).AddStart(
+			func(ctx execution.Context) {
+				assert.Equal(t, objectGlobal, ctx.Object)
+				assert.Equal(t, contract.CallIntolerable, ctx.Isolation.Interference)
+				synchronizeExecution.WakeUp()
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("second result"), objectGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify(firstOutgoing.String(), intolerableFlags(), nil)
+		runnerMock.AddExecutionClassify(secondOutgoing.String(), intolerableFlags(), nil)
+	}
+
+	// add typedChecker mock
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	{
+		typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+			switch result.CallOutgoing {
+			case firstOutgoing:
+				assert.Equal(t, []byte("first result"), result.ReturnArguments)
+			case secondOutgoing:
+				assert.Equal(t, []byte("second result"), result.ReturnArguments)
+			default:
+				t.Fatalf("unexpected outgoing")
+			}
+			return false
+		})
+	}
+
+	// 2 Tolerable VCallRequests will be converted to Intolerable
+	{
+		executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 2)
+
+		// send first
+		pl := &payload.VCallRequest{
+			CallType:            payload.CTMethod,
+			CallFlags:           payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+			Caller:              server.GlobalCaller(),
+			Callee:              objectGlobal,
+			CallSiteDeclaration: class,
+			CallSiteMethod:      "GetMethod",
+			CallOutgoing:        firstOutgoing,
+		}
+		server.SendPayload(ctx, pl)
+		// wait for the first
+		commontestutils.WaitSignalsTimed(t, 10*time.Second, synchronizeExecution.Wait())
+		// send second
+		pl = &payload.VCallRequest{
+			CallType:            payload.CTMethod,
+			CallFlags:           payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+			Caller:              server.GlobalCaller(),
+			Callee:              objectGlobal,
+			CallSiteDeclaration: class,
+			CallSiteMethod:      "GetMethod",
+			CallOutgoing:        secondOutgoing,
+		}
+		server.SendPayload(ctx, pl)
+		commontestutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
+	}
+
+	// wait for all VCallResults
+	commontestutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+	assert.Equal(t, 2, typedChecker.VCallResult.Count())
+
+	synchronizeExecution.Done()
```

Comment:
I'd suggest to use defer for `synchronizeExecution.Done() `, it can leave hanging goroutines behind